tests/tests/test-class-db-driver-wpdb.php

```diff
@@ -98,6 +111,190 @@ public function test_table_names() {
+	public function test_purge_storage_multisite() {
+		global $wpdb;
+
+		if ( ! is_multisite() ) {
+			$this->markTestSkipped( 'This test requires multisite.' );
+		}
+
+		// Test cases override table dropping queries to prevent them
+		// from being executed, but we need actual dropping here.
+		foreach ( $GLOBALS['wp_filter']['query'] as $priority => $filters ) {
+			foreach ( $filters as $filter => $data ) {
+				if ( false !== strpos( $filter, '_drop_temporary_table' ) ) {
+					unset( $filters[$filter] );
+					$GLOBALS['wp_filter']['query'][$priority] = $filters;
+				}
+			}
+		}
+
+		$this->assertNotEmpty( $this->driver->table );
+		$stream_result = $wpdb->get_results( "SHOW TABLES LIKE '{$this->driver->table}'", ARRAY_A );
+		$this->assertNotEmpty( $stream_result );
+		$this->assertNotEmpty( $this->driver->table_meta );
+		$stream_meta_result = $wpdb->get_results( "SHOW TABLES LIKE '{$this->driver->table_meta}'", ARRAY_A );
+		$this->assertNotEmpty( $stream_meta_result );
+
+		// Ensure the tables for the current site are not empty.
+		$blog_id = get_current_blog_id();
+		$stream_ids = $wpdb->get_var( $wpdb->prepare( "SELECT ID FROM {$this->driver->table} WHERE site_id = %d", $blog_id ) );
+		$this->assertGreaterThan( 0, count( $stream_ids ) );
+		foreach ( $stream_ids as $stream_id ) {
+			$stream_meta_count = $wpdb->get_var( $wpdb->prepare( "SELECT COUNT(*) FROM {$this->driver->table_meta} WHERE record_id = %d", $stream_id ) );
+			$this->assertGreaterThan( 0, $stream_meta_count );
+		}
+
+		// Admin user access is needed to uninstall the plugin.
+		$admin_user = $this->factory->user->create( array( 'role' => 'administrator' ) );
+		wp_set_current_user( $admin_user );
+
+		// Trigger purge operation directly.
+		$uninstall = $this->driver->purge_storage( wp_stream_get_instance() );
+		$_REQUEST['nonce'] = wp_create_nonce( 'stream_uninstall_nonce' );
+
+		$uninstall->uninstall();
+
+		// On multisite, the tables are not deleted, but the records are.
+		$stream_result = $wpdb->get_results( "SHOW TABLES LIKE '{$this->driver->table}'", ARRAY_A );
+		$this->assertNotEmpty( $stream_result );
+		$stream_meta_result = $wpdb->get_results( "SHOW TABLES LIKE '{$this->driver->table_meta}'", ARRAY_A );
+		$this->assertNotEmpty( $stream_meta_result );
+
+		$stream_count = $wpdb->get_var( $wpdb->prepare( "SELECT COUNT(*) FROM {$this->driver->table} WHERE site_id = %d", $blog_id ) );
+		$this->assertEquals( 0, $stream_count );
+		$stream_id_placeholders = implode( ', ', array_fill( 0, count( $stream_ids ), '%d' ) );
+		$stream_meta_count = $wpdb->get_var( $wpdb->prepare( "SELECT COUNT(*) FROM {$this->driver->table_meta} WHERE record_id IN ( $stream_id_placeholders )", $stream_ids ) );
+		$this->assertEquals( 0, $stream_meta_count );
+	}
+
+	// Test that the purge_storage() function can only be called with admin capabilities.
+	// TODO: This throws a notice: "Test code or tested code did not (only) close its own output buffers".
```

Comment:
The tests that throw an exception here are flagged as `risky` because they seem to break the output buffer logic somewhere. However, I have failed to pinpoint the cause of this as of yet.

Reply:
This was due to the `catch()` wrappers which are not needed for the `_handleAjax()` helper and the associated logic that relies on catching the `dieHandler` override here:

https://github.com/wp-phpunit/wp-phpunit/blob/a21378376bbec9366c4b18a77d62f42a7dfdabac/includes/testcase-ajax.php#L216-L228